From 897d40d4e37b7eb5039b075edb9b708ee86d1e0f Mon Sep 17 00:00:00 2001
From: Vadim Pirogov <vadim.o.pirogov@intel.com>
Date: Wed, 14 Oct 2026 19:36:12 -0700
Subject: [PATCH 2/2] build: changed /link to /Qoption,link

All flags after /link are passed directly to linker. This behavior
unintentionally drops -fsycl flag during linking.
---
 cmake/SDL.cmake | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmake/SDL.cmake b/cmake/SDL.cmake
index fd3e0f27a4..cb950bc3f3 100644
--- a/cmake/SDL.cmake
+++ b/cmake/SDL.cmake
@@ -92,7 +92,7 @@ elseif(WIN32)
         endif()
     elseif(CMAKE_BASE_NAME STREQUAL "icx")
         append(ONEDNN_SDL_COMPILER_FLAGS "/EHs /GS /Gy /guard:cf /Wformat /Wformat-security")
-        append(ONEDNN_SDL_LINKER_FLAGS "/link /NXCOMPAT")
+        append(ONEDNN_SDL_LINKER_FLAGS "/Qoption,link,/NXCOMPAT")
     elseif(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
         append(ONEDNN_SDL_COMPILER_FLAGS "-Wformat -Wformat-security")
         if(UPPERCASE_CMAKE_BUILD_TYPE STREQUAL "RELEASE")
@@ -117,7 +117,7 @@ elseif(WIN32)
         # in the Safe Load List.
         if(CMAKE_BASE_NAME STREQUAL "icx")
             # add ICX-style linker flags
-            append(ONEDNN_SDL_LINKER_FLAGS "/link /DEPENDENTLOADFLAG:0x2000")
+            append(ONEDNN_SDL_LINKER_FLAGS "/Qoption,link,/DEPENDENTLOADFLAG:0x2000")
         elseif(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
             # add Clang-style linker flags
             append(ONEDNN_SDL_LINKER_FLAGS "-Xlinker /DEPENDENTLOADFLAG:0x2000")
-- 
2.43.0

